backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -242,27 +245,28 @@ public IGitMacheteRepositorySnapshot discoverLayoutAndCreateSnapshot() throws Gi
       return result;
     }
 
-    Option<String> inferParentForLocalBranch(
-        Set<String> eligibleBranchNames,
+    Option<ILocalBranchReference> inferParentForLocalBranch(
+        Set<String> eligibleLocalBranchNames,
         String localBranchName) throws GitCoreException {
 
-      var branch = localBranchByName.get(localBranchName).getOrNull();
-      if (branch == null) {
+      var localBranch = localBranchByName.get(localBranchName).getOrNull();
+      if (localBranch == null) {
         return Option.none();
       }
 
       LOG.debug(() -> "Branch(es) eligible for becoming the parent of ${localBranchName}: " +
-          "${eligibleBranchNames.mkString(\", \")}");
-      String remoteTrackingBranchName = branch.getRemoteTrackingBranch().map(rtb -> rtb.getName()).getOrNull();
+          "${eligibleLocalBranchNames.mkString(\", \")}");
 
       var commitAndContainingBranches = gitCoreRepository
-          .ancestorsOf(branch.getPointedCommit())
+          .ancestorsOf(localBranch.getPointedCommit())
           .map(commit -> {
-            Seq<String> eligibleContainingBranches = deriveBranchesContainingGivenCommitInReflog()
+            Seq<ILocalBranchReference> eligibleContainingBranches = deriveBranchesContainingGivenCommitInReflog()
                 .getOrElse(commit.getHash(), List.empty())
-                .filter(candidateBranchName -> !candidateBranchName.equals(branch.getName())
-                    && !candidateBranchName.equals(remoteTrackingBranchName)
-                    && eligibleBranchNames.contains(candidateBranchName));
+                .map(candidateBranch -> candidateBranch.isLocal()
+                    ? candidateBranch.asLocal()
+                    : candidateBranch.asRemote().getTrackedLocalBranch())
+                .filter(correspondingLocalBranch -> !correspondingLocalBranch.getName().equals(localBranch.getName())
+                    && eligibleLocalBranchNames.contains(correspondingLocalBranch.getName()));
```

Comment:
This is the actual fix

Reply:
If a commit (ancestor of commit point by branch X) has been found only in reflog of origin/Y but not Y itself, let's still consider Y as a viable parent for X